SG14/ring.h

```diff
@@ -10,21 +10,22 @@ namespace sg14
 	template <typename T>
 	struct null_popper
 	{
-		void operator()(T&);
+		void operator()(T&) const noexcept;
 	};
 
 	template <typename T>
 	struct default_popper
 	{
-		T operator()(T& t);
+		T operator()(T& t) const;
 	};
 
 	template <typename T>
 	struct copy_popper
 	{
-		copy_popper(T&& t);
-		T operator()(T& t);
-		T copy;
+		explicit copy_popper(T t);
```

Comment:
The `explicit` broke one of the tests included here: https://github.com/WG21-SG14/SG14/commit/aba97661a03d99712d87bba61c1d4fd03569eb05#commitcomment-22430301